server/apiserver/argoserver.go

```diff
@@ -334,7 +334,10 @@ func (as *argoServer) newHTTPServer(ctx context.Context, port int, artifactServe
 		grpc.WithDefaultCallOptions(grpc.MaxCallRecvMsgSize(MaxGRPCMessageSize)),
 	}
 	if as.tlsConfig != nil {
-		dialOpts = append(dialOpts, grpc.WithTransportCredentials(credentials.NewTLS(as.tlsConfig)))
+		tlsConfig := as.tlsConfig
+		tlsConfig.InsecureSkipVerify = true
```

Comment:
@chtcvl Do you need to make InsecureSkipVerify=true for all cases? Can you make as command line arg and configure in `tsConfig`?

Reply:
Right now `gRPC server` explicitly runs on `localhost` ( https://github.com/argoproj/argo-workflows/blob/769896eb5bf0a7d8db1a94b423e5bc16cf09d5aa/server/apiserver/argoserver.go#L260 ) meaning `https://localhost`. For `grpc-gateway` to access `gRPC server` it has some options:
1 . set `InsecureSkipVerify=true` (not really a concern as this is communication between two goroutines in the same app) - let grpc-gateway trust gRPC server running on localhost. 
2.  have certificate Subject Alternate Name include `localhost` - this is not supported by publicly trusted certificates. 
3. serve on https://realdomain... for which certificate is issued instead of localhost.

This being said, adding new option for `InsecureSkipVerify=false` can lead to confusion as option will only be usable in pt.3 which is not supported by current code.

Reply:
Thanks for your response. LGTM @alexec Can you quickly take a look? if I am missing anything.